hazelcast/include/hazelcast/client/protocol/ClientMessage.h

```diff
@@ -795,6 +875,201 @@ class HAZELCAST_API ClientMessage
         return h;
     }
 
+    /**
+     * Reads the header of the current frame.
+     * The cursor must be at a frame's beginning.
+     */
+    frame_header_t read_frame_header()
+    {
+        frame_header_t header{};
+        auto pos = rd_ptr(SIZE_OF_FRAME_LENGTH_AND_FLAGS);
+        std::memcpy(&header.frame_len, pos, sizeof(header.frame_len));
+        pos += sizeof(header.frame_len);
+        std::memcpy(&header.flags, pos, sizeof(header.flags));
+        return header;
+    }
+
+    template<typename T>
+    typename std::enable_if<std::is_same<T, sql::sql_column_metadata>::value,
+                            T>::type
+    get()
+    {
+        // skip begin frame
+        skip_frame();
+
+        const frame_header_t header = read_frame_header();
+
+        auto type = static_cast<sql::sql_column_type>(get<int32_t>());
```

Comment:
What happens if this integer is not a valid `sql_column_type` ? What kind of exception is thrown? Is it understandable easily? In Java a clear error is thrown. https://github.com/hazelcast/hazelcast/blob/84d0749c90bc844a29f912dc10a3f5777db767a9/hazelcast/src/main/java/com/hazelcast/client/impl/protocol/codec/builtin/CustomTypeFactory.java#L237

Reply:
good catch. Fixed at commit https://github.com/hazelcast/hazelcast-cpp-client/pull/995/commits/03b6894b20de719475f92e6ab4fb600b6ec30186